### ISSUE TITLE
IQKeyboardReturnKeyHandler strong reference to delegate

### ISSUE THREAD

**Issue by dvanoni** [hackiftekhar/IQKeyboardManager#760]
I'm using the Swift version of `IQKeyboardReturnKeyHandler` in a view controller that adopts `UITextFieldDelegate` and `UITextViewDelegate`. I'm setting the `delegate` of my `IQKeyboardReturnKeyHandler` to this view controller so that it can perform a custom action when handling `textFieldShouldReturn`.

My first attempt, which works, looked like this:
```swift
var returnKeyHandler: IQKeyboardReturnKeyHandler?

override func viewDidLoad() {
    super.viewDidLoad()
    
    returnKeyHandler = IQKeyboardReturnKeyHandler(controller: self)
    returnKeyHandler?.delegate = self
}
```

However, because `IQKeyboardReturnKeyHandler` holds a _strong_ reference to `delegate`, the above code contains a retain cycle and the view controller is never deallocated.

My new approach to avoid this retain cycle is as follows:
```swift
override func viewWillAppear(_ animated: Bool) {
    super.viewWillAppear(animated)
    returnKeyHandler?.delegate = self
}

override func viewDidDisappear(_ animated: Bool) {
    super.viewDidDisappear(animated)
    returnKeyHandler?.delegate = nil
}
```

I was wondering if it makes sense to change the following [line][code] to a _weak_ reference to avoid this issue?
```swift
open var delegate: UITextFieldDelegate & UITextViewDelegate?
```

[code]: https://github.com/hackiftekhar/IQKeyboardManager/blob/master/IQKeyboardManagerSwift/IQKeyboardReturnKeyHandler.swift#L40

**Comment by hackiftekhar** [comment]
@dvanoni Thanks a lot for pointing out. Happy coding.

**Comment by dvanoni** [comment]
Thanks for the fix!

**Comment by feniixx** [comment]
Is this fixed for v5.0.3? Or do we have to do:

`override func viewWillAppear(_ animated: Bool) {
    super.viewWillAppear(animated)
    returnKeyHandler?.delegate = self
}

override func viewDidDisappear(_ animated: Bool) {
    super.viewDidDisappear(animated)
    returnKeyHandler?.delegate = nil
}`

**Comment by hackiftekhar** [comment]
The issue mentioned was already fixed.
